### PR TITLE
added how to guide on sequencing tasks

### PR DESCRIPTION
**Pull request by eed3si9n**

### CODE REVIEW

File: src/reference/04-Howto/20-Howto-Sequencing/00.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+This is more constrained compared to the imperative style plain Scala code with side effects such as the follows:
+
+```scala
+def foo(): Unit = {
+  doX()
+  doY()
+}
+```
+
+The benefit of the dependency-oriented programming model is that sbt's task engine is able to reorder the task execution. When possible we run dependent tasks in parallel. Another benefit is that we can deduplicate the graph, and make sure that the `compile` is called once per command execution.
```
Comment (dwijnand) [inline src/reference/04-Howto/20-Howto-Sequencing/00.md]: do you mean `compile` is special-cased in the task engine?

**Comment by dwijnand** [comment]
LGTM!

I have one question, but mostly for my own benefit, not a blocker.

File: src/reference/04-Howto/20-Howto-Sequencing/04-Howto-Dynamic-Input-Task.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  settings(
+    run in Compile := (Def.inputTaskDyn {
+      import sbt.complete.Parsers.spaceDelimited
+      val args = spaceDelimited("<args>").parsed
+      Def.taskDyn {
+        (actualRun in Compile).toTask(" " + args.mkString(" ")).value
+        openbrowser
+      }
+    }).evaluated,
+    actualRun in Compile := Defaults.runTask(
```
Comment (smarter) [inline src/reference/04-Howto/20-Howto-Sequencing/04-Howto-Dynamic-Input-Task.md]: It'd be helpful if this tutorial explained how you came up with the definition of `actualRun in Compile`.

File: src/reference/04-Howto/20-Howto-Sequencing/00.md (hunk 1 above)
Comment (eed3si9n) [inline src/reference/04-Howto/20-Howto-Sequencing/00.md]: no all tasks are deduplicated.

File: src/reference/04-Howto/20-Howto-Sequencing/00.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+This is more constrained compared to the imperative style plain Scala code with side effects such as the follows:
+
+```scala
+def foo(): Unit = {
+  doX()
+  doY()
+}
+```
+
+The benefit of the dependency-oriented programming model is that sbt's task engine is able to reorder the task execution. When possible we run dependent tasks in parallel. Another benefit is that we can deduplicate the graph, and make sure that the task evaluation, such as `compile in Compile`, is called once per command execution, as oppose to compiing the same source many times.
```
Comment (dwijnand) [inline src/reference/04-Howto/20-Howto-Sequencing/00.md]: s/compiing/compiling/

Comment (dwijnand) [inline src/reference/04-Howto/20-Howto-Sequencing/00.md]: s/oppose/opposed/

**Pull request merged**